tests/cupyx_tests/scipy_tests/signal_tests/test_ltisys.py

```diff
@@ -540,3 +541,318 @@ def test_second_order(self, xp, scp):
+                      0.6750, 1.067, 4.273, -6.654, 5.893, 0.0480, 4.273,
+                      1.343, -2.104]).reshape(4, 4)
+        B = xp.array([0, 5.679, 1.136, 1.136, 0, 0, -3.146, 0]).reshape(4, 2)
+
+        fsf = scp.signal.place_poles(A, B, xp.asarray([2, 2, 3, 3]))
+        return xp.real_if_close(fsf.computed_poles)
+
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_complex(self, xp, scp):
+        # Test complex pole placement on a linearized car model, taken from L.
+        # Jaulin, Automatique pour la robotique, Cours et Exercices, iSTE
+        # editions p 184/185
+        A = xp.array([[0, 7, 0, 0],
+                      [0, 0, 0, 7/3.],
+                      [0, 0, 0, 0],
+                      [0, 0, 0, 0]])
+        B = xp.array([[0, 0],
+                      [0, 0],
+                      [1, 0],
+                      [0, 1]])
+        # Test complex poles on YT
+        P = xp.array([-3, -1, -2-1j, -2+1j])
+
+        fsf = scp.signal.place_poles(A, B, P)
+        return fsf.computed_poles
+
+    @testing.numpy_cupy_allclose(scipy_name='scp', atol=1e-8)
+    def test_complex_2(self, xp, scp):
+        # Try to reach the specific case in _YT_complex where two singular
+        # values are almost equal. This is to improve code coverage but I
+        # have no way to be sure this code is really reached
+        A = xp.array([[0, 7, 0, 0],
+                      [0, 0, 0, 7/3.],
+                      [0, 0, 0, 0],
+                      [0, 0, 0, 0]])
+        B = xp.array([[0, 0],
+                      [0, 0],
+                      [1, 0],
+                      [0, 1]])
+        P = xp.array([0-1e-6j, 0+1e-6j, -10, 10])
+
+        fsf = scp.signal.place_poles(A, B, P, maxiter=1000)
+        return fsf.computed_poles
+
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_complex_3(self, xp, scp):
+        # Try to reach the specific case in _YT_complex where the rank two
+        # update yields two null vectors. This test was found via Monte Carlo.
+
+        A = xp.array(
+            [-2148, -2902, -2267, -598, -1722, -1829, -165, -283, -2546,
+             -167, -754, -2285, -543, -1700, -584, -2978, -925, -1300,
+             -1583, -984, -386, -2650, -764, -897, -517, -1598, 2, -1709,
+             -291, -338, -153, -1804, -1106, -1168, -867, -2297]
+        ).reshape(6, 6)
+
+        B = xp.array(
+            [-108, -374, -524, -1285, -1232, -161, -1204, -672, -637,
+             -15, -483, -23, -931, -780, -1245, -1129, -1290, -1502,
+             -952, -1374, -62, -964, -930, -939, -792, -756, -1437,
+             -491, -1543, -686]
+        ).reshape(6, 5)
+        P = xp.array([-25.-29.j, -25.+29.j, 31.-42.j,
+                     31.+42.j, 33.-41.j, 33.+41.j])
+
+        fsf = scp.signal.place_poles(A, B, P)
+        return fsf.computed_poles
+
+    @pytest.mark.xfail(reason="numerical stability: scipy QR vs numpy QR")
+    @testing.numpy_cupy_allclose(scipy_name='scp')
```

Comment:
Does relaxing the tolerance (`rtol` and `atol` of `numpy_cupy_allclose`) remove the xfail?

Reply:
Not really. This test is flaky, and failure modes depend on the scipy version. Comments in the scipy code indicate things start differing between scipy.linalg.qr and numpy.qr (?), so this whole function is a bit brittle.
So the last commit skips the test depending on the scipy version, based on what the CI seems to suggest.